core/types/validator.go

```diff
@@ -87,3 +88,24 @@ func (set *ValidatorSet) Contains(addr common.Address) bool {
 	_, ok := set.membership[addr]
 	return ok
 }
+
+func NewDeposit(amount *big.Int, unixTimestamp int64) *Deposit {
```

Comment:
this would sit nice under Validator. Seems like a good option for an Aggregate with Validator as top level

https://martinfowler.com/bliki/DDD_Aggregate.html